cheri_compressed_cap_common.h

```diff
@@ -67,22 +67,32 @@ _CC_STATIC_ASSERT_SAME(_CC_N(FIELD_INTERNAL_EXPONENT_SIZE) + _CC_N(FIELD_EXP_NON
                            _CC_N(FIELD_EXPONENT_LOW_PART_SIZE),
                        _CC_N(FIELD_EBT_SIZE));
 
+// In order to allow vector loads and store from memory we can optionally reverse the first two fields.
 struct _cc_N(cap) {
     /* offset = cursor - base */
+#ifdef _CC_REVERSE_PESBT_CURSOR_ORDER
+    /* Currently only updated lazily when compress is called.
```

Comment:
Do you mean that, or do you mean setbounds? Doing it on compress is going to break memcpy's if it recanonicalises.

Reply:
Stale comment (have fixed). I think keeping pesbt always is sync with every other field is best.

Reply:
Yeah I strongly agree it should always be in sync. Not doing that caused numerous bugs in CHERI-MIPS QEMU where weird things would happen depending on whether you stored or not.

Reply:
I dislike that "lazily" is used, it's not for optimisation purposes, it's for correctness (and not really _lazy_ either).